### PR TITLE
Movesmsbackends

### PR DESCRIPTION
**Pull request by dannyroberts**
@gcapalbo I think we talked about this a while back, and I tested out that at the very least tests pass.

I did this by running [`./moveapps`](https://github.com/dimagi/commcare-hq/commit/a87e1a6c0006f0414d6dc5788927aa9c1ebe3fbe) (which I committed and then removed after using, so that there's a record of it). The following apps will now live in `corehq/messaging/smsbackends/`:

```
grapevine
megamobile
sislog
telerivet
tropo
twilio
unicel
yo
mach
kookoo
```

What do you think? Do you think this will break anything—like dot-separated import paths saved to the DB or something? FWIW, I did this using a subtree merge which means that (I believe) any outstanding PRs touching any of these apps can be automatically merged with this respecting the file rename.

### CODE REVIEW

File: corehq/messaging/smsbackends/unicel/urls.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.conf.urls import *
+
+urlpatterns = patterns('',
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/unicel/urls.py]: W291 trailing whitespace

File: corehq/messaging/smsbackends/unicel/urls.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.conf.urls import *
+
+urlpatterns = patterns('',         
+    url(r'^in/$', 'corehq.messaging.smsbackends.unicel.views.incoming'),
+)
+
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/unicel/urls.py]: W391 blank line at end of file

**Comment by gcapalbo** [comment]
@dannyroberts Thanks, this organization makes good sense. There's just one backend (kookoo) that still uses the import path in the db doc, so I can update that in both our environments once deployed.

I noticed that the original directories (twilio, tropo, etc.) under corehq/apps/ are still there - was that intentional?

Before we move forward with this, I'm also going to add some tests to this branch to make sure we didn't break anything.

**Comment by dannyroberts** [comment]
> I noticed that the original directories (twilio, tropo, etc.) under corehq/apps/ are still there - was that intentional?

I believe git has no way to empty delete directories, so when you switch to a branch that has a directory and then switch back, it leaves the empty directory there.

About the doc in the db, another thing that you could presumably do is switch from using `to_function` to defining a static mapping. So like,

```python
from corehq.messaging.smsbackends.kookoo import foo_function

FUNCTION_MAPPING = {
    'corehq.apps.kookoo.foo_function': foo_function
}
```
etc. I'm also fine with your offer of doing it manually after the fact, depending on the way these things are used.

**Comment by dannyroberts** [comment]
@gcapalbo any thoughts or progress on your last comment? Would be nice to get this in soon or decide to punt it

**Comment by gcapalbo** [comment]
@dannyroberts I haven't gotten a chance yet to test this out and I want to give it the appropriate attention to make sure we don't have any issues.   I should be able to finish this up in a week or so.  Does that sound ok?

**Comment by dannyroberts** [comment]
Yup, sounds great, thanks!

**Comment by dimagimon** [comment]
There are 10 errors:

* corehq/apps/sms/tests/test_all_backends.py, line 2 - F401 'TestCase' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 3 - F401 'Client' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 9 - F401 'UnicelBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 10 - F401 'MachBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 11 - F401 'TropoBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 12 - F401 'HttpBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 13 - F401 'TelerivetBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 14 - F401 'TestSMSBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 15 - F401 'GrapevineBackend' imported but unused
* corehq/apps/sms/tests/test_all_backends.py, line 17 - F401 'MegamobileBackend' imported but unused

File: corehq/apps/sms/tests/test_all_backends.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+from corehq.messaging.smsbackends.tropo.api import TropoBackend
+from corehq.messaging.smsbackends.http.api import HttpBackend
+from corehq.messaging.smsbackends.telerivet.models import TelerivetBackend
+from corehq.apps.sms.test_backend import TestSMSBackend
+from corehq.messaging.smsbackends.grapevine.api import GrapevineBackend
+from corehq.messaging.smsbackends.twilio.models import TwilioBackend
+from corehq.messaging.smsbackends.megamobile.api import MegamobileBackend
+from dimagi.utils.parsing import json_format_datetime
+from django.conf import settings
+from django.test import TestCase
```
Comment (dimagimon) [inline corehq/apps/sms/tests/test_all_backends.py]: F401 'TestCase' imported but unused

**Comment by gcapalbo** [comment]
@dannyroberts Ok this is good to go now from my side

@nickpell Fyi, the sms backends now exist in a different directory. Are you ok with this update from the billing side / does the billing code have tests that would catch any issues this might cause?

**Comment by dannyroberts** [comment]
thanks @gcapalbo, +1 on the tests. @nickpell we'll wait on your reply to merge

**Comment by nickpell** [comment]
This is fine by the billing side - the smsbillables gateway fee tests should catch any issues.

**Comment by gcapalbo** [comment]
ok, thanks @nickpell for confirming

**Comment by dannyroberts** [comment]
:tada:

**Pull request merged**